src/repository/adaptive-fetching-strategy.ts

```diff
@@ -0,0 +1,127 @@
+import { EventEmitter } from 'events';
+import { UnleashEvents } from '../events';
+import { Mode } from '../unleash-config';
+import { FetchingStrategyInterface, FetchingStrategyOptions } from './fetching-strategy';
+import { PollingStrategy } from './polling-strategy';
+import { StreamingStrategy } from './streaming-strategy';
+
+export class AdaptiveFetchingStrategy extends EventEmitter implements FetchingStrategyInterface {
+  private currentStrategy: FetchingStrategyInterface;
+
+  private pollingStrategy: PollingStrategy;
```

Comment:
Both polling and streaming strategies are created upfront rather than on-demand to enable fast mode switching while preserving state (etag, failure counts) across transitions